R/createQuadrantGate.R

```diff
@@ -0,0 +1,95 @@
+#'   populations must refer.
+#' @param gids Group IDs of each sector, assigned to `model.gid`.
+#' @param parentPopulationId ID of the parent population. Use \code{NULL} for
+#'   the "ungated" population. If specified, do not specify \code{parentPopulation}.
+#' @param parentPopulation Name of the parent population. An attempt will be made
+#'   to find the population by name. If zero or more than one population exists
+#'   with the name, an error will be thrown. If specified, do not specify
+#'   \code{parentPopulationId}.
+#' @param tailoredPerFile Whether this gate is tailored per FCS file.
+#' @param fcsFileId ID of FCS file, if tailored per file. Use \code{NULL} for
+#'   the global gate in a tailored gate group. If specified, do not specify
+#'   \code{fcsFile}.
+#' @param fcsFile Name of FCS file, if tailored per file. An attempt will be made
+#'   to find the file by name. If zero or more than one file exists with the name,
+#'   an error will be thrown. If specified, do not specify \code{fcsFileId}.
+#' @param locked Prevents modification of the gate via the web interface.
+#' @param createPopulation Automatically create corresponding population.
+#' @export
+#' @examples
+#' \dontrun{
+#' createQuadrantGate(experimentId, "FSC-A", "FSC-W", "my gate", 160000, 200000)
+#' }
+createQuadrantGate = function(experimentId, xChannel, yChannel, name,
+                               x, y, angles = c(0.000000, 3/2*pi, pi, pi/2),
+                               gid = generateId(), gids = replicate(4, generateId()),
+                               parentPopulationId = NULL, parentPopulation = NULL, skewable = FALSE,
+                               tailoredPerFile = FALSE, fcsFileId = NULL, fcsFile = NULL,
+                               locked = FALSE, createPopulation = TRUE) {
+
+    scales = data.frame(getScaleSets(experimentId)$scales)
+    labels = list(
+               c(scales[scales$channelName==xChannel, ]$scale$maximum, # upper right
+               scales[scales$channelName==yChannel, ]$scale$maximum), # upper right
+               c(scales[scales$channelName==xChannel, ]$scale$minimum, # upper left
+               scales[scales$channelName==yChannel, ]$scale$maximum), # upper left
+               c(scales[scales$channelName==xChannel, ]$scale$minimum, # lower left
+               scales[scales$channelName==yChannel, ]$scale$minimum),  # lower left
+               c(scales[scales$channelName==xChannel, ]$scale$maximum, # lower right
+               scales[scales$channelName==yChannel, ]$scale$minimum)  # lower right
+               )
+
+
+    #TODO(ge): vectorize this with foreach or itertools packages?
+    add_suffixes = function(name){
+      suffixes = c("(UR)", "(UL)", "(LL)", "(LR)")
+      i = 1
+      all_names = c()
+      for (suffix in suffixes){
+        all_names = append(all_names, paste(name, suffixes[i]))
+        i = i+1
+      }
+      return(all_names)
+    }
+
+    names = add_suffixes(name)
```

Comment:
Would it work to do something like `lapply(c("(UR)", "(UL)", "(LL)", "(LR)"), function (suffix) paste(name, suffix))`?